src/app/presence-control/services/presence-control-state.service.ts

```diff
@@ -312,6 +317,17 @@ export class PresenceControlStateService
                 : [presence];
             }
           }, [] as Array<LessonPresence>)
+          .map((presence) =>
+            getPresenceControlEntry(
+              extractLessonEntries(presences).find(
+                (lesson) => lesson.id === presence.LessonRef.Id.toString()
+              ),
+              presence,
+              types,
+              confirmationStates,
+              otherTeachersAbsences
+            )
+          )
```

Comment:
Hier machst du das `extractLessonEntries` für jede Iteration im `map`. Würde das vorrangig assignen, da das ja nicht ändert:

```
map(([presences, types, confirmationStates, otherTeachersAbsences]) =>
  const lessons = extractLessonEntries(presences);
  return presences
    .filter(...)
    .map((presence) =>
      getPresenceControlEntry(
        lessons.find(...),
        ...
      )
    );
});

Reply:
Noch besser, wir machen ja bereits ein `extractLessonEntries` beim `lessions$`. Du könntest also hier `this.lessions$.pipe(take(1))` mit reingeben, dann kannst du auf das `const lessons = extractLessonEntries(presences);` verzichten.